runtime/jvmti/jvmtiHook.c

```diff
@@ -3631,3 +3640,31 @@ jvmtiHookVmDumpEnd(J9HookInterface** hook, UDATA eventNum, void* eventData, void
 
 	TRACE_JVMTI_EVENT_RETURN(jvmtiHookVmDumpEnd);
 }
+
+/**
+ * Check if a JVMTI event should be dispatched by checking the flag set by
+ * notifyJvmtiHideFrames() and the presence of the JvmtiMountTransition
+ * annotation.
+ *
+ * Note that the method should only be checked for Method Entry and Method Exit event.
+ *
+ * @param currentThread the current thread to be checked
+ * @param method the method to be checked
+ * @return TRUE if the event should be posted
+ */
+static BOOLEAN
+shouldPostEvent(J9VMThread *currentThread, J9Method *method)
+{
+#if JAVA_SPEC_VERSION >= 20
+	BOOLEAN shouldPost = J9_ARE_NO_BITS_SET(currentThread->privateFlags, J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES);
+	if (NULL != method) {
+		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+		U_32 extendedModifiers = getExtendedModifiersDataFromROMMethod(romMethod);
+		shouldPost = shouldPost && J9_ARE_NO_BITS_SET(extendedModifiers, CFR_METHOD_EXT_JVMTIMOUNTTRANSITION_ANNOTATION);
+	}
+	return shouldPost;
+#else /* JAVA_SPEC_VERSION >= 20 */
+	return TRUE;
+#endif /* JAVA_SPEC_VERSION >= 20 */
+
```

Comment:
This can be improved; my first observation was `shouldPost = shouldPost && ...`. Suggest:
```c
	BOOLEAN shouldPost = TRUE;
#if JAVA_SPEC_VERSION >= 20
	if (J9_ARE_ANY_BITS_SET(currentThread->privateFlags, J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES)) {
		shouldPost = FALSE;
	} else if (NULL != method) {
		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
		U_32 extendedModifiers = getExtendedModifiersDataFromROMMethod(romMethod);

		if (J9_ARE_ANY_BITS_SET(extendedModifiers, CFR_METHOD_EXT_JVMTIMOUNTTRANSITION_ANNOTATION)) {
			shouldPost = FALSE;
		}
	}
#endif /* JAVA_SPEC_VERSION >= 20 */
	return shouldPost;
```
Note that this would yield the right answer even if the `#if` / `#endif` lines were removed.